feat: 22878 activate physical card

MrMuzyk opened the pull request:
### Details
Draft PR for [22878](https://github.com/Expensify/App/issues/22878)

### Fixed Issues
<!---
1. Please postfix `$` with a URL link to the GitHub issue this Pull Request is fixing. For example, `$ https://github.com/Expensify/App/issues/<issueID>`.
2. Please postfix  `PROPOSAL:` with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).  For example, `PROPOSAL: https://github.com/Expensify/App/issues/<issueID>#issuecomment-1369752925`

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking and its automation will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<issueID>
$ https://github.com/Expensify/App/issues/<issueID(comment)>

Do NOT only link the issue number like this: $ #<issueID>
--->
$ 
PROPOSAL: 


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

- [ ] Verify that no errors appear in the JS console

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [ ] I linked the correct issue in the `### Fixed Issues` section above
- [ ] I wrote clear testing steps that cover the changes made in this PR
    - [ ] I added steps for local testing in the `Tests` section
    - [ ] I added steps for the expected offline behavior in the `Offline steps` section
    - [ ] I added steps for Staging and/or Production testing in the `QA steps` section
    - [ ] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [ ] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I ran the tests on **all platforms** & verified they passed on:
    - [ ] Android / native
    - [ ] Android / Chrome
    - [ ] iOS / native
    - [ ] iOS / Safari
    - [ ] MacOS / Chrome / Safari
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [ ] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] I verified that if a function's arguments changed that all usages have also been updated correctly
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Mobile Web - Chrome</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Mobile Web - Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Desktop</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Android</summary>

<!-- add screenshots or videos here -->

</details>

rezkiy37 commented on src/components/MagicCodeInput.js:
```diff
@@ -48,6 +48,12 @@ const propTypes = {
     maxLength: PropTypes.number,
+
+    /** Specifies if the keyboard should be disabled */
+    disableKeyboard: PropTypes.bool,
```
We prefer `isSomething` or `shouldSomething`, when a value is boolean. Also, `disableKeyboard` sounds like a function to do disable.
```suggestion
    isKeyboardDisabled: PropTypes.bool,
```

MrMuzyk replied:
This is literally what `TextInput` component widely used in the app expects. Prop name is 1:1 the same

rezkiy37 replied:
Yeah, I see. I just remember - https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#boolean-variables-and-props. This case could be an exception.

MrMuzyk replied:
k, ill change it

rezkiy37 commented on src/components/MagicCodeInput.js:
```diff
@@ -237,6 +245,27 @@ function MagicCodeInput(props) {
+        onChangeText(value);
+
+        // We have not added:
+        // + the onChangeText and onKeyPress as the dependencies because we only want to run this when lastPressedDigit changes.
```
Can we wrap `onChangeText` into a `useCallback` to memorise the function?

gedu replied:
Given that `onChangeText` isn't being use into any dependency array, there is no need to wrap it into a `useCallback`

rezkiy37 replied:
@gedu, `onChangeText` is being used in `useEffect` and explicitly skipped in a deps. array of this `useEffect`.

gedu replied:
as the comment said `only want to run this when lastPressedDigit changes`, so if the function isn't being used in the dependency array, there is no need to use `useCallback`, because the hook `useEffect` isn't checking if the `onChangeText` change the reference.

rezkiy37 replied:
Yeah, makes sense. I just thought that we can a full deps. array. Forget.

rezkiy37 commented on src/pages/settings/Wallet/ActivateCardPage.js:
```diff
@@ -0,0 +1,175 @@
+    /* Onyx Props */
+    /** The details about the Expensify cards */
+    cardList: PropTypes.shape({
+        isLoading: PropTypes.bool,
+        [PropTypes.string]: PropTypes.objectOf(assignedCardPropTypes),
+        [PropTypes.string]: PropTypes.shape({
+            // eslint-disable-next-line react/forbid-prop-types
+            errors: PropTypes.object,
+        }),
```
1. Can we expect when there is a shape of cards, we always have `isLoading`? 
2. Can we extend a card object?

```suggestion
        isLoading: PropTypes.bool.isRequired,
        [PropTypes.string]: PropTypes.objectOf({
            ...assignedCardPropTypes,   
            errors: PropTypes.object,
        }),
```

MrMuzyk replied:
![image](https://github.com/pasyukevich/Expensify-app/assets/23487612/4eea8180-bb85-4e2c-83c9-649fb0ef8f0d)

This is what ill be expecting in onyx. Errors are going to be said on the object under the cardID key. Not sure if there is a nicer way to shape props in such scenario.

As for isLoading, I agree

rezkiy37 replied:
Ah, I see. Therefore, no sense to extend a card object with errors.

MrMuzyk replied:
I had to re-do the props completely because it turns out that the structure I was expecting was incorrect. Also after a second thought, isLoading cant be required prop. Its not gonna be there in the begining, its added after 1st API call

rezkiy37 replied:
Thank you for letting me know 🙂

rezkiy37 commented on src/pages/settings/Wallet/ActivateCardPage.js:
```diff
@@ -0,0 +1,175 @@
+    return (
+        <IllustratedHeaderPageLayout
+            title={translate('activateCardPage.activateCard')}
+            onBackButtonPress={() => Navigation.goBack()}
```
```suggestion
            onBackButtonPress={Navigation.goBack}
```

rezkiy37 commented on src/pages/settings/Wallet/ActivateCardPage.js:
```diff
@@ -0,0 +1,175 @@
+                />
+            </View>
+            <View style={[styles.w100, styles.justifyContentEnd, styles.pageWrapper]}>
+                {DeviceCapabilities.canUseTouchScreen() ? <BigNumberPad numberPressed={updateLastPressedDigit} /> : null}
```
If `canUseTouchScreen` always returns a boolean value, we can simplify a condition.
```suggestion
                {DeviceCapabilities.canUseTouchScreen() && <BigNumberPad numberPressed={updateLastPressedDigit} />}
```

adhorodyski replied:
Honestly I'd default to a ternary - the underlying implementation might change and it will affect this component causing a regression.

rezkiy37 replied:
It is already migrated to TS and has a [strict type](https://github.com/Expensify/App/blob/main/src/libs/DeviceCapabilities/canUseTouchScreen/types.ts). So, we can expect that it returns boolean forever. However, it is not something critical. Up to you.

rezkiy37 commented on src/pages/settings/Wallet/ExpensifyCardPage.js:
```diff
@@ -16,10 +16,18 @@ import * as CurrencyUtils from '../../../libs/CurrencyUtils';
 const propTypes = {
     /* Onyx Props */
-    cardList: PropTypes.objectOf(assignedCardPropTypes),
+    cardList: PropTypes.shape({
```
This value is being used a few times. Let's create a file with a prop types for `cardList` to re-use it across files. Also, default props can we in a file. Example https://github.com/Expensify/App/blob/main/src/components/categoryPropTypes.js.

rezkiy37 commented on src/pages/settings/Wallet/ExpensifyCardPage.js:
```diff
@@ -89,7 +100,16 @@ function ExpensifyCardPage({
                         )}
+                        {}
                     </ScrollView>
+                    {physicalCard.state === CARD_NOT_ACTIVATED_STATE ? (
```
```suggestion
                    {physicalCard.state === CARD_NOT_ACTIVATED_STATE && (
```

adhorodyski replied:
@rezkiy37 I often like to refer to this Kent's article https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx

rezkiy37 replied:
I got you. Also, makes sense.

MrMuzyk replied:
I guess its accepted here, there is even a point in the PR checklist - ` I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. myBool && <MyComponent />.`

rezkiy37 commented on src/pages/settings/Wallet/ExpensifyCardPage.js:
```diff
@@ -89,7 +100,16 @@ function ExpensifyCardPage({
+                            onPress={() => Navigation.navigate(ROUTES.SETTINGS_WALLET_CARD_ACTIVATE.getRoute(domain))}
+                            text={translate('activateCardPage.activatePhysicalCard')}
+                        />
+                    ) : null}
```
```suggestion
                    )}
```

rezkiy37 reviewed (commented):
Left comments.

pasyukevich commented on src/types/onyx/CardList.ts:
```diff
@@ -0,0 +1,10 @@
+import {Errors} from './OnyxCommon';
+
+type CardList = Record<string, {errors?: Errors}> & {
+    physical: Card;
```
are you sure that we will have such keys? 

As far as I know, we will use here card ID as a key

gedu commented on src/pages/settings/Wallet/ActivateCardPage.js:
```diff
@@ -0,0 +1,175 @@
+const CARD_ACTIVATED_STATE = 3;
+
+function ActivateCardPage({
+    cardList,
```
It is a list or an object? I think the name is confusing

MrMuzyk replied:
I wasnt the one who created this onyxkey, it was already in the codebase added by some1 else

BeeMargarida commented on src/libs/actions/CardSettings.js:
```diff
@@ -0,0 +1,57 @@
+ * @param {Number} lastFourDigits
+ * @param {Number} cardID
+ */
+
```
unnecessary new line

MrMuzyk replied:
shouldnt it be covered by prettier?

BeeMargarida replied:
It should, but I think it's not

BeeMargarida commented on src/pages/settings/Wallet/ActivateCardPage.js:
```diff
@@ -0,0 +1,175 @@
+    }),
+
+    /** Navigation route context info provided by react navigation */
+    route: PropTypes.shape({
```
I would move route as the first prop, since it's not an Onyx prop. It seems like this is the standard.

MrMuzyk replied:
After looking into it I dont think its a standard. ExpensifyCardPage just got merged and it doesnt follow this practice. Neither do plenty of other components

BeeMargarida commented on src/pages/settings/Wallet/ActivateCardPage.js:
```diff
@@ -0,0 +1,175 @@
+    cardList: {},
+};
+
+const ACTIVATE_CARD_CODE_DESIRED_LENGTH = 4;
+const CARD_ACTIVATED_STATE = 3;
```
Can't this be moved to CONST? The same with the const in ExpensifyCardPage.
I think it would be useful to have all the states under a readable const, grouped by ACTIVE_CARD or something like that. Like the `STATE_NUM` const of a report.

pasyukevich commented on src/pages/settings/Wallet/ActivateCardPage.js:
```diff
@@ -0,0 +1,175 @@
+            return;
+        }
+
+        if (!cardList.isLoading && lodashGet(cardList, 'physical.state', 0) === CARD_ACTIVATED_STATE) {
```
pls use  `CardUtils.getDomainCards(cardList)[domain];`

we don't need whole list here, only the domain

pasyukevich replied:
https://docs.google.com/document/d/1rFxJ78vz5On6zSWzYa51B9v-tyLdC5pUsBeLOLig0t4/edit#heading=h.s5rzg79l078b

pasyukevich commented on src/pages/settings/Wallet/ActivateCardPage.js:
```diff
@@ -0,0 +1,175 @@
+     * If state of the card is CARD_ACTIVATED_STATE, navigate to card details screen.
+     */
+    useEffect(() => {
+        if (cardList.isLoading) {
```
Can we put isLoading state to the related card / domain ?

pasyukevich commented on src/ONYXKEYS.ts:
```diff
@@ -342,7 +342,7 @@ type OnyxValues = {
     [ONYXKEYS.BANK_ACCOUNT_LIST]: Record<string, OnyxTypes.BankAccount>;
     [ONYXKEYS.FUND_LIST]: Record<string, OnyxTypes.Fund>;
-    [ONYXKEYS.CARD_LIST]: Record<string, OnyxTypes.Card>;
+    [ONYXKEYS.CARD_LIST]: OnyxTypes.CardList;
```
I think we need to have here previous structure and apply changes to the `Card` entity itself

pasyukevich commented on src/libs/actions/CardSettings.js:
```diff
@@ -0,0 +1,57 @@
+ * @param {Number} cardID
+ */
+
+function activatePhysicalExpensifyCard(lastFourDigits, cardID) {
```
I'm a bit confused with namings in declaring and calling 

Here I see lastFourDigits argument, but in the call we have activateCardCode.

What naming is describing this value?

rezkiy37 reviewed (approved):
LGTM.

pasyukevich commented on src/pages/settings/Wallet/ExpensifyCardPage.js:
```diff
@@ -10,16 +10,22 @@ import CardPreview from '../../../components/CardPreview';
+    /** The details about the Expensify cards */
+    cardList: PropTypes.shape({
+        isLoading: PropTypes.bool,
+        [PropTypes.number]: PropTypes.objectOf(assignedCardPropTypes),
```
In Onyx we will have cardList object with  key value: 

```
{
   'string' : assignedCardPropTypes,
   'anotherstring' : assignedCardPropTypes,
   ....
}
```

Here you updated to such structure: 

```
{
   isLoading: PropTypes.bool,
   23452345: {
   'string' : assignedCardPropTypes,
   'anotherstring' : assignedCardPropTypes,
   ....
   }
}
```

pasyukevich commented on src/types/onyx/CardList.ts:
```diff
@@ -0,0 +1,7 @@
+import Card from './Card';
+
+type CardList = Record<number, Card> & {
```
We don't have number key in current structure from BE

pasyukevich commented on src/types/onyx/CardList.ts:
```diff
@@ -0,0 +1,7 @@
+import Card from './Card';
+
+type CardList = Record<number, Card> & {
+    isLoading: boolean;
```
Can we move this to the Card itself?

pasyukevich commented on src/pages/settings/Wallet/ExpensifyCardPage.js:
```diff
@@ -89,7 +95,16 @@ function ExpensifyCardPage({
                         )}
+                        {}
                     </ScrollView>
+                    {physicalCard.state === CONST.CARD_STATE.NOT_ACTIVATED && (
```
Looks like this part is not workable anymore since we do not have `physicalCard` variable here

pasyukevich commented on src/libs/actions/CardSettings.js:
I guess CardSettings is not suitable here, since activation is not part of card settings 

What do you think about  CardActions.js ?

pasyukevich replied:
Or even simply Card.js

pasyukevich commented on src/pages/settings/Wallet/ExpensifyCardPage.js:
```diff
@@ -10,16 +10,21 @@ import CardPreview from '../../../components/CardPreview';
     /* Onyx Props */
-    cardList: PropTypes.objectOf(assignedCardPropTypes),
+    /** The details about the Expensify cards */
+    cardList: PropTypes.shape({
```
In this case we can have just `cardList: PropTypes.objectOf(assignedCardPropTypes),`

pasyukevich commented on src/pages/settings/Wallet/ActivateCardPage.js:
```diff
@@ -0,0 +1,179 @@
+    /* Onyx Props */
+
+    /** The details about the Expensify cards */
+    cardList: PropTypes.shape({
```
In this case we can have just `cardList: PropTypes.objectOf(assignedCardPropTypes),`

coleaeason committed:
Set ruby version to avoid 3.2 default or system default

coleaeason committed:
Remove ruby version, fix liquid requirement

coleaeason committed:
Set char set in scss

coleaeason committed:
Revert utf change

shubham1206agra committed:
Merge branch 'main' into fix-persisted-requests

roryabraham committed:
Merge pull request #28886 from Expensify/cole_rbenv

Setup Cloudflare Pages for Docs Preview

youssef-lr committed:
Merge pull request #27632 from dukenv0307/fix/25947

Fix auto scroll when the display name is changed

NikkiWines committed:
Merge pull request #28389 from getusha/fix-scroll-behind-suggestion-list

Fix scroll behind suggestion list when scrolling on suggestion popup

dukenv0307 committed:
fix route of edit receipt request

aldo-expensify committed:
Merge pull request #28675 from namhihi237/fix-28195-remove-validate-blur-password-pdf

remove validate when blur

Victor-Nyagudi committed:
Remove part that says Expensify doesn't use Prettier

pecanoro committed:
Merge pull request #28639 from getusha/fix-hover-lost-opening-popover-again

Fix: Hover effect lost after opening popover again before closing it

thienlnam committed:
Merge pull request #28513 from Expensify/strepanier03-patch-3

Julesssss committed:
Merge pull request #27952 from Pujan92/fix/26419

Fix: Approved requests shouldn't have a delete option

Julesssss committed:
Merge pull request #28745 from Expensify/marklouisdeshaun-patch-1

Update Consolidated-Domain-Billing.md

Julesssss committed:
Merge pull request #28891 from Victor-Nyagudi/update-readme-for-prettier

Remove part that says Expensify doesn't use Prettier in README

arosiclair committed:
Merge pull request #28683 from tienifr/fix/28354

fix: 28354 Report screen becomes empty when resizing to big size from LHN

arosiclair committed:
Merge pull request #28702 from dukenv0307/fix/28087

Fix LHN is disappeared when leaving thread

suneox committed:
update isLoadingApp key instead of isLoading

laurenreidexpensify committed:
Update Attendee-Tracking.md

sangar-1028 committed:
fix: conflicts

sangar-1028 committed:
fix: conflicts

DylanDylann committed:
fix pending field when edit task

DylanDylann committed:
fix pending field when edit task assignee

Julesssss committed:
add dedicated screenshot/video section to standard issue template

Julesssss committed:
move details to top of issue for better visibility

Julesssss committed:
align the pull request device list with the standard issue template

Julesssss committed:
reintroduce mWeb for clarity

Julesssss committed:
align device naming across all issue templates

bernhardoj committed:
fix wrong onyx key

adamgrzybowski committed:
adjustments v5

rezkiy37 committed:
add AUTO_REPORTING_FREQUENCIES

rezkiy37 committed:
add SUBMITTED

MrMuzyk committed:
Merge branch 'main' of github.com:MrMuzyk/App into feat-22878-activate-physical-card

wildan-m committed:
Merge branch 'main' of https://github.com/wildan-m/App into wildan/fix/26648-3-tab-hover

MrMuzyk committed:
Merge branch 'main' of github.com:MrMuzyk/App into migrateOnfidoWebsiteToFunctional

wildan-m committed:
fix tab opacity

grgia committed:
move down

adamgrzybowski committed:
fix color for focused item in global navigation

mountiny committed:
Merge pull request #23910 from margelo/fix/follow-up-regression-form-checboxes

[NoQA] Check for boolean input values

MrMuzyk committed:
fix: cr fixes

grgia committed:
Merge branch 'main' into georgia-cardTransactions-HelperFunctions

huzaifa-99 committed:
Merge branch 'main' into 25855-get-user-location

wildan-m committed:
extract style logic outside the style object

mountiny committed:
Merge pull request #28890 from dukenv0307/fix/27965-regression

[CP Staging] Fix route of edit receipt request

adamgrzybowski committed:
fix top elements aligment

OSBotify committed:
Update version to 1.3.78-1

BeeMargarida committed:
feat: show submit button in Report preview for open/draft reports

robertjchen committed:
Merge pull request #28646 from akinwale/task-27607

fix: room name field does not automatically focus when returning from the chat option

Nikhil-Vats committed:
Merge branch 'main' into 26070_fix_popover_position

BeeMargarida committed:
refactor: remove brackets

adamgrzybowski committed:
change ref for GlobalNavigationMenuItem

BeeMargarida committed:
docs: add comment with TODO

MaciejSWM committed:
Migrate to Typescript

wildan-m committed:
run prettier

MaciejSWM committed:
Extract returned value to type

robertjchen committed:
Merge pull request #27420 from tienifr/fix/26657

fix: 26657 Another error message for a short time when opening archived chat from search bar

wildan-m committed:
add activeOpacity, inactiveOpacity to tabOpacity style function

wildan-m committed:
run prettier

tgolen committed:
Merge pull request #26546 from huzaifa-99/25855-get-user-location

Added option to get user's current location

aldo-expensify committed:
Merge pull request #28714 from MrMuzyk/migrateOnfidoNativeToFunctionComponent

feat: migrate onfido native class component to functional

MariaHCD committed:
Merge pull request #28788 from Expensify/nkuoch-patch-1

Add images for HelpDot Reimbursing Reports

dangrous committed:
Merge pull request #28451 from ZhenjaHorbach/fix/distance-replace-popup-closes-after-image-on-esc

johnmlee101 committed:
Merge pull request #28843 from abzokhattab/fix/private-notes-raw-text-links

Fix private notes raw text links issue

mountiny committed:
Revert "fix: use the same sort order for header display names and avatars in a group chat view"

rezkiy37 committed:
remove extra props

wildan-m committed:
unify tabOpacity style function

srikarparsi committed:
Merge pull request #28929 from Expensify/revert-28218-task-28048

[CP Staging] Revert "fix: use the same sort order for header display names and avatars in a group chat view"

wildan-m committed:
Revert "fix tab opacity"

This reverts commit 3b9681688f808b085239df146f73782568f81a67.

OSBotify committed:
Update version to 1.3.78-2

puneetlath committed:
Merge pull request #28680 from hungvu193/fix-27568

Disable flash when there is no permission granted

AndrewGable committed:
Merge pull request #28845 from DylanDylann/fix/28007-city-field

Fix: City field is lost when login again by deeplink

Nikhil-Vats committed:
Add safety check for target element

VickyStash committed:
Merge branch 'main' into ts-migration/activeClientManager-lib

mountiny committed:
Merge pull request #28815 from MrMuzyk/migrateOnfidoWebsiteToFunctional

feat: migrate onfido web component from class to functioncal

neil-marcellini committed:
Fix, show address suggestions of all types

Nikhil-Vats committed:
Improve comment

tgolen committed:
Merge pull request #28938 from Expensify/neil-suggest-places

Fix show waypoint address suggestions of all types

Nikhil-Vats committed:
Add empty line before comment

waterim committed:
add submit action

mountiny committed:
Merge pull request #28931 from rezkiy37/fix/28919-disabled-fields-on-confirm-task

[CP Staging] Fix disabled chevron of "MenuItem"

mkhutornyi committed:
import moment for es locale

waterim committed:
lint fix

marcochavezf committed:
Merge pull request #28744 from Nikhil-Vats/26070_fix_popover_position

Fix popover position for payment buttons

mkhutornyi committed:
prevent focusing on composer while popover is open

OSBotify committed:
Update version to 1.3.78-3

mountiny committed:
Revert "Merge pull request #27836 from Expensify/dangrous-roomwelcomemessages"

This reverts commit 447d87e9204578ccaf5c8e3a6887b97de531457e, reversing
changes made to 7f063c7b6487e037804e1d97c08a739071e67679.

Krishna2323 committed:
fix: Workspaces - Cursor opacity is not reduced when checkbox is disabled.

Signed-off-by: Krishna Gupta <belivethatkg@gmail.com>

NicMendonca committed:
Update docs/articles/expensify-classic/getting-started/Plan-Types.md

Co-authored-by: Rushat Gabhane <rushatgabhane@gmail.com>

tylerkaraszewski committed:
Merge pull request #28378 from DylanDylann/fix/25956

Add pending fields when edit task assignee

AndrewGable committed:
Merge pull request #27035 from shubham1206agra/fix-25595

Fixed no result found after selecting a address from the drop-down menu

marcaaron committed:
Merge pull request #28674 from s-alves10/fix/issue-27445

fix: keep tailing spaces in heading

mountiny committed:
Merge pull request #28955 from Expensify/vit-revert27836

[CP Staging] Revert "Merge pull request #27836 from Expensify/dangrous-roomwelcomemessages"

allroundexperts committed:
fix: handle comment suggestions

OSBotify committed:
Update version to 1.3.78-4

allroundexperts committed:
fix: remove the setValidateCode call while resending the magic code

stitesExpensify committed:
Merge branch 'main' of github.com:Expensify/App into NicMendonca-patch-2

stitesExpensify committed:
Merge pull request #26234 from pasyukevich/migrate-ts/focusAndUpdateMultilineInputRange

[No QA] [TS migration] Migrate 'focusAndUpdateMultilineInputRange.js' lib to TypeScript

marcaaron committed:
Merge pull request #28757 from allroundexperts/fix-28553

fix: clear the input when resending validate code

mountiny committed:
Update Adhoc Provisioning profile

bondydaa committed:
Merge pull request #28868 from SrR0b0/fix/28074

fix: room avatars attachment modal shouldn't close when avatar image is updated

Commit:
Merge pull request #28394 from Expensify/bfitzexpensify-patch-4

Create connect-ANZ.md

Commit:
Merge pull request #27918 from VickyStash/ts-migration/pusherUtils-lib

[No QA] [TS migration] Migrate 'PusherUtils.js' lib to TypeScript

MitchExpensify committed:
Create Xero Integration HelpDoc Article 

Coming from here, this intended to match the changes described here https://github.com/Expensify/Expensify/issues/311510

techievivek committed:
Merge pull request #28676 from Expensify/RachCHopkins-patch-2

Update FinancalForce.md

puneetlath committed:
Merge pull request #28213 from azamuddin/fix/proptypes-warning-policy-required-money-request-header

fix: mark propTypes policy as optional and provides default value

Li357 committed:
Merge pull request #28306 from kubabutkiewicz/ts-migration/IntlPolyfill-lib

[No QA] [TS migration] Migrate 'IntlPolyfill' lib to TypeScript

gadhiyamanan committed:
Fix opens two flag as offensive list

MariaHCD committed:
Merge pull request #27792 from VickyStash/ts-migration/activeClientManager-lib

[No QA] [TS migration] Migrate 'ActiveClientManager' lib to TypeScript

rojiphil committed:
last message fetch using optimistic actions

Commit:
feat: add new comp for iframe

madmax330 committed:
Merge pull request #28812 from graylewis/Gray-ShareCode-404

Initial fix

OSBotify committed:
Update version to 1.3.79-0

tsa321 committed:
add optimistic data to archive workspace chats  when remove members of policies

Signed-off-by: Tsaqif <tsaiinkwa@yahoo.com>

Julesssss committed:
Merge pull request #28755 from Expensify/joekaufmanexpensify-patch-1

Adding Expensify Card dispute help site article

Commit:
feat: convert NewDot URL to OldDot URL

aldo-expensify committed:
Merge pull request #28681 from DylanDylann/fix/27779-expand-button-disappear-after-clicking

Fix/27779: Expand button disappear after clicking send

OSBotify committed:
Update version to 1.3.79-1

tsa321 committed:
run prettier

Signed-off-by: Tsaqif <tsaiinkwa@yahoo.com>

abzokhattab committed:
Fixing zoom and google items being accessible after room achieving

grgia committed:
add isExpensifyCard

grgia committed:
Merge branch 'main' into georgia-cardTransactionsTS

grgia committed:
Merge pull request #28246 from Expensify/georgia-cardTransactionsTS

[ECard Transactions] Update `Transactions` and `Report` TS Types

grgia committed:
Merge branch 'main' into georgia-cardTransactions-HelperFunctions

OSBotify committed:
Update version to 1.3.79-2

MrMuzyk committed:
feat: update layout

grgia committed:
add hasEreceipt

grgia committed:
add getEReceiptColor

OSBotify committed:
Update version to 1.3.79-3

grgia committed:
add getCardDescription

grgia committed:
prettier

Julesssss committed:
Merge pull request #28796 from paultsimura/fix/27536-thread-reaction

fix: Reactions in thread do not update parent action

Beamanator committed:
Merge pull request #25543 from hoangzinh/df/22041

Fix incorrect navigation when update Report welcome message after visit by direct link

MonilBhavsar committed:
Merge pull request #28567 from ishpaul777/fix/not-able-delete-digit-before-decimal

fix: update add leading zero to delete digit before decimal

thienlnam committed:
Merge pull request #28443 from dukenv0307/fix/28030

Beamanator committed:
Merge pull request #28387 from rezkiy37/fix/18681-inconsistent-margins

Fix inconsistent margins for reacted emojis to code-block comments

thienlnam committed:
Merge pull request #28842 from suneox/fix/28077-fix-display-name-on-first-load

grgia committed:
Add MCC icons

grgia committed:
Add missing Utilities icon

grgia committed:
update spanish translations from review in slack

dukenv0307 committed:
Update src/components/PopoverWithoutOverlay/index.js

Co-authored-by: Rajat Parashar <parasharrajat@users.noreply.github.com>

grgia committed:
fix lint

grgia committed:
fix comments

mountiny committed:
Merge pull request #28810 from shubham1206agra/fix-persisted-requests

Fix: RBR - Deleted error notifications reappear after reloading

grgia committed:
prettier for lint

esh-g committed:
Merge branch 'main' into disable-checkbox

mountiny committed:
Merge pull request #28972 from Expensify/vit-updateAdHocProvisioningProfile

[NoQA] Update Adhoc Provisioning profile

Commit:
Merge pull request #28980 from rojiphil/27948-lhn-update-oftaskparent

last message fetch using optimistic actions

BeeMargarida committed:
fix: check if report is expense and change button color

MrMuzyk committed:
fix: bigger inputs, dry code and undefined cardID

mountiny committed:
Merge pull request #28175 from waterim/feat-19810-Replace-other-places

[NoQA] feat: Remove moment from the project (Except Datepicker)

techievivek committed:
Merge pull request #28992 from abzokhattab/fixing-zoom-and-google-items-being-shown-after-archeiving-rooms

Fixing zoom and google items being accessible after room achieving

grgia committed:
change to backgroundColor and color

grgia committed:
Add Ereceipt BG file

BeeMargarida committed:
fix: use chat report

rezkiy37 committed:
Merge branch 'main' of https://github.com/rezkiy37/Expensify into feature/28759-new-report-consts

rezkiy37 committed:
extend Policy.ts

rezkiy37 committed:
create OriginalMessageSubmitted of ts

deetergp committed:
Merge pull request #28785 from BhuvaneshPatil/27703-align-checkboxes

fix: align checkbox with input levels

AndrewGable committed:
Merge pull request #28993 from Expensify/georgia-cardTransactions-HelperFunctions

[No QA]Add card transactions helper functions and mcc icons

mountiny committed:
Merge branch 'main' into vit-accessNotificationSafely

mountiny committed:
Access the actionName safely

MitchExpensify committed:
Update docs/articles/new-expensify/integrations/accounting-integrations/Xero

Co-authored-by: Rocio Perez-Cano <pecanoro@users.noreply.github.com>

mountiny committed:
Merge pull request #28918 from BeeMargarida/feat/28766-submit_button_report_preview

[NoQA] feat: show submit button in Report preview for open/draft reports

neil-marcellini committed:
Merge pull request #28948 from mkhutornyi/fix-28315

import moment for es locale

Selyatin committed:
Fix lint by running prettier

Li357 committed:
Merge pull request #28511 from Expensify/strepanier03-patch-2

Update Invoicing.md

francoisl committed:
Merge pull request #28978 from gadhiyamanan/fix-28863

cristipaval committed:
Merge pull request #28905 from bernhardoj/fix/28466-load-more-skeleton

Fix load more message skeletons aren't showing

tylerkaraszewski committed:
Merge pull request #28882 from tsa321/fouroct

Fix not all split bills are registered when leaving offline mode

cristipaval committed:
Merge pull request #27385 from Selyatin/fix-sign-in-not-working-when-not-using-leader-tab

Show notification instead of login page when user has another tab open - #26161

chiragsalian committed:
Merge pull request #28957 from Krishna2323/krishna2323/issue/28597

fix: Workspaces - Cursor opacity is not reduced when checkbox is disabled

wildan-m committed:
Resolve issue hover not working after pressing back from another screen

deetergp committed:
Merge pull request #28146 from zukilover/webfix/26529-unread-task

Fix unread marker showing double lines on task comment

Commit:
Merge pull request #28277 from adamgrzybowski/@swm/global-nav-menu-v1

@swm/global nav menu v1

dukenv0307 committed:
add explaination comment

wildan-m committed:
Merge branch 'main' of https://github.com/wildan-m/App into wildan/fix/26648-3-tab-hover

wildan-m committed:
tidy up code

wildan-m committed:
run prettier

dukenv0307 committed:
fix lint

dukenv0307 committed:
Merge branch 'main' into fix/28241

dukenv0307 committed:
add first render ref

dukenv0307 committed:
update comment

dukenv0307 committed:
Update src/components/PopoverWithoutOverlay/index.js

Co-authored-by: Rajat Parashar <parasharrajat@users.noreply.github.com>

dukenv0307 committed:
Update src/components/PopoverWithoutOverlay/index.js

Co-authored-by: Rajat Parashar <parasharrajat@users.noreply.github.com>

dukenv0307 committed:
fix lint

mountiny committed:
Merge pull request #28880 from Expensify/vit-accessNotificationSafely

[NoQA] Access the report action safely in notification logic

waterim committed:
adjustments

mountiny committed:
Merge pull request #28615 from kacper-mikolajczak/chat-transition-optimisation-proposal

Improve memoization of LHN by handling the isFocused prop better

pecanoro committed:
Merge pull request #28975 from Expensify/MitchExpensify-patch-1

Create Xero Integration HelpDoc Article

Commit:
fix: get URL from the window

Commit:
fix: silence lint error

dukenv0307 committed:
resolve conflict

pecanoro committed:
Merge pull request #28645 from Expensify/tgolen-remove-sideloading-editrequestpage

Refactor how data is loaded into the edit request page

pecanoro committed:
Merge pull request #28728 from Expensify/laurenreidexpensify-patch-5

Update Attendee-Tracking.md

Commit:
Merge pull request #28397 from Expensify/bfitzexpensify-patch-7

Create Add-a-Deposit-Account-(AUD).md

MariaHCD committed:
Merge pull request #28357 from wildan-m/wildan/fix/26648-3-tab-hover

Add Hover to TabItem

Commit:
fix: the logical OR

Commit:
Merge pull request #28981 from Expensify/hayata-add-iframe-comp

Implement Iframe component

kosmydel committed:
Merge branch 'Expensify:main' into @kosmydel/es-country-flags

nkuoch committed:
Merge pull request #28746 from Expensify/marklouisdeshaun-patch-2

Update Tax-Exempt.md

ZhenjaHorbach committed:
Fix Delete request pop-up does not close when requestee pays the money request

dukenv0307 committed:
fix comments

dukenv0307 committed:
change comment

rezkiy37 committed:
Merge branch 'main' of https://github.com/rezkiy37/Expensify into fix/26506-hide-category-in-split-bill

rezkiy37 committed:
Merge branch 'main' of https://github.com/rezkiy37/Expensify into feature/28759-new-report-consts

rezkiy37 committed:
clarify comments

rezkiy37 committed:
sort consts

Skalakid committed:
Merge branch 'main' into @Skalakid/ts/NetworkConnection

waterim committed:
merge main, resolve conflicts

ZhenjaHorbach committed:
Fix comment

rezkiy37 committed:
simplify combining array

mountiny committed:
Merge pull request #28377 from dukenv0307/fix/27759-catch-translation-key

puneetlath committed:
Merge pull request #27936 from rezkiy37/fix/26506-hide-category-in-split-bill

Allow Adding a category to a Split Bill

lakchote committed:
Merge pull request #28850 from esh-g/disable-checkbox

Disable SelectCircle for current user in split bill

Commit:
feat: set authToken and email cookies from NewDot

tsa321 committed:
add failure data for report actions and add default value for oldPolicy failure data

Signed-off-by: Tsaqif <tsaiinkwa@yahoo.com>

Commit:
docs: fix comment style

eh2077 committed:
Fix: adding parentheses after link removes link

amyevans committed:
Merge pull request #28903 from Expensify/jules-addCaptureSectionToStandardIssuetemplate

[NO QA] Add screenshot/video section to standard issue template

waterim committed:
adjustments

waterim committed:
Merge remote-tracking branch 'upstream/main' into feat/28763-submitReport-IOU-action

waterim committed:
fixes

mountiny committed:
Merge pull request #28947 from waterim/feat/28763-submitReport-IOU-action

[NoQA] Feat: Add submit action

mountiny committed:
Merge pull request #28987 from tsa321/archwchats

Add optimistic data to archive workspace chats when remove members

deetergp committed:
Merge pull request #28486 from dukenv0307/fix/28241

Fix navigation breaks on opening search page after reloading the attachment

rezkiy37 committed:
Merge branch 'main' of https://github.com/rezkiy37/Expensify into feature/28759-new-report-consts

rezkiy37 committed:
Merge branch 'main' of https://github.com/rezkiy37/Expensify into feature/28593-viewing-category-split-bill

mountiny committed:
Merge pull request #28906 from rezkiy37/feature/28759-new-report-consts

[NoQA] Add the "auto reporting" and "submitted" state as const to the App

lakchote committed:
Merge pull request #28985 from eh2077/28344-fix-adding-parentheses-after-link-removes-link

Fix: adding parentheses after link removes link

puneetlath committed:
Merge pull request #28628 from rezkiy37/feature/28593-viewing-category-split-bill

Allow Viewing a category to a Split Bill

puneetlath committed:
Merge pull request #26200 from kosmydel/@kosmydel/es-country-flags

Fix country flags not searchable in spanish language

Li357 committed:
Merge pull request #28679 from Expensify/RachCHopkins-patch-3

Update QuickBooks-Desktop.md

amyevans committed:
Revert "fix: 26657 Another error message for a short time when opening archived chat from search bar"

chiragsalian committed:
Merge pull request #29106 from Expensify/revert-27420-fix/26657

[CP Staging] Revert "fix: 26657 Another error message for a short time when opening archived chat from search bar"

OSBotify committed:
Update version to 1.3.79-4

mountiny committed:
Merge pull request #26358 from hungvu193/fix-23234

Fix Chat - Notification is NOT received in the main device after Control policy was created

amyevans committed:
Merge pull request #28953 from mkhutornyi/fix-25485

prevent focusing on composer while popover is open

dangrous committed:
Merge pull request #28348 from Expensify/NicMendonca-patch-2

jasperhuangg committed:
Revert "Initial fix"

jasperhuangg committed:
Merge pull request #29126 from Expensify/revert-28812-Gray-ShareCode-404

Revert "Initial fix"

srikarparsi committed:
Merge pull request #28920 from software-mansion-labs/ts-migration/useThrottledButtonState

[TS migration] Migrate 'useThrottledButtonState.js' hook to TypeScript

OSBotify committed:
Update version to 1.3.79-5

OSBotify committed:
Update version to 1.3.80-0

robertjchen committed:
Merge pull request #28016 from software-mansion-labs/@Skalakid/ts/NetworkConnection

[TS migration] Migrate 'NetworkConnection.js' lib to TypeScript

Commit:
fix: remove selector

robertjchen committed:
Merge pull request #29089 from Expensify/hayata-pass-auth-token-to-embedded-olddot

Set `authToken` and `email` cookies from NewDot

MonilBhavsar committed:
Merge pull request #29068 from ZhenjaHorbach/delete-request-pop-up-does-not-close

Delete request pop-up does not close when requestee pays the money request

MrMuzyk committed:
Merge branch 'main' of github.com:MrMuzyk/App into feat-22878-activate-physical-card

MrMuzyk committed:
fix: removed card state const and swap to new added const, reverted change in onyxkeys for cardlist
